cypher/cypher-docs/src/docs/dev/query-tuning.asciidoc

```diff
@@ -125,6 +126,95 @@ In Enterprise Edition, the Cypher query planner selects the runtime, falling bac
+| `connectComponentsPlanner=idp`
+| Use the cost based IDP search algorithm when combining sub-plans.
+  Note that using this option may significantly increase the planning time of the query. |
+|===
+
+[[cypher-update-strategy]]
+=== Cypher update strategy
+This option affects eagerness of updating queries
+[options="header"]
+|===
+| Query option | Description | Default
+| `updateStrategy=default` | Update queries are executed eagerly when needed. | X
+| `updateStrategy=eager`   | Update queries are always executed eagerly. |
+|===
+
+[[cypher-expression-engine]]
+=== Cypher expression engine
+This option affects how the runtime evaluates expressions. The possible values are:
+[options="header"]
+|===
+| Query option | Description | Default
+| `expressionEngine=default`     | Compile expressions and use the compiled expression engine when needed. | X
+| `expressionEngine=interpreted` | Always use the interpreted expression engine. |
+| `expressionEngine=compiled`    | Always compile expressions and use the compiled expression engine. Cannot be used together with `runtime=interpreted` |
+|===
+
+[[cypher-operator-engine]]
+=== Cypher operator engine
+This query option affects whether the pipelined runtime attempts to fuse groups of operators into pipelines. The possible values are:
+[options="header"]
+|===
+| Query option | Description | Default
+| `operatorEngine=default`     | Attempt to fuse operators when applicable. | X
+| `operatorEngine=interpreted` | Never attempt to fuse operators. |
+| `operatorEngine=compiled`    | Always attempt to fuse operators. Cannot be used together with `runtime=interpreted` or `runtime=slotted` |
+|===
+
+[[cypher-interpreted-pipes-fallback]]
+=== Cypher interpreted pipes fallback
+This query option affects how the pipelined runtime behaves for operators it does not directly support. The available options are:
+[options="header"]
+|===
+| Query option | Description | Default
+| `interpretedPipesFallback=default`
+| Equivalent to `interpretedPipesFallback=whitelisted_plans_only`
+| X
+| `interpretedPipesFallback=disabled`
+| If the plan contains any operators not supported by the pipelined runtime then another runtime is chosen to execute the entire plan.
+  Cannot be used together with `runtime=interpreted` or `runtime=slotted`
+|
+| `interpretedPipesFallback=whitelisted_plans_only`
+| Parts of the execution plan may be executed on another runtime.
+  Only certain operators are allowed to execute on another runtime.
+  Cannot be used together with `runtime=interpreted` or `runtime=slotted`
+|
+| `interpretedPipesFallback=all`
+| Parts of the execution plan may be executed on another runtime.
+  Any operator is allowed to execute on another runtime.
+  This setting is experimental and using it is discouraged.
+  Queries with this option set might produce incorrect results or fail.
+  Cannot be used together with `runtime=interpreted` or `runtime=slotted`
+|
+|===
```

Comment:
Having each cell on a new line reads a bit weird when looking at the bare asciidoc. I would prefer having each table row correspond to a line of source code.

Reply:
Well, the lines get really long.. The same thing is done in the `cypher-runtime`section above

Reply:
I can clean up the asciidoc, but this will render as a single row in the table, as per the other tables on this page. 

I'm just reviewing now, and will push some editorial changes.